crates/polars-stream/src/physical_plan/lower_ir.rs

```diff
@@ -17,13 +17,50 @@ pub fn lower_ir(
     expr_arena: &mut Arena<AExpr>,
     phys_sm: &mut SlotMap<PhysNodeKey, PhysNode>,
 ) -> PolarsResult<PhysNodeKey> {
-    match ir_arena.get(node) {
+    let ir_node = ir_arena.get(node);
+    match ir_node {
         IR::Filter { input, predicate } if is_streamable(predicate.node(), expr_arena) => {
             let predicate = predicate.clone();
             let input = lower_ir(*input, ir_arena, expr_arena, phys_sm)?;
             Ok(phys_sm.insert(PhysNode::Filter { input, predicate }))
         },
 
+        // TODO: split partially streamable selections to avoid fallback as much as possible.
+        IR::Select {
+            input,
+            expr,
+            schema,
+            ..
+        } if expr.iter().all(|e| is_streamable(e.node(), expr_arena)) => {
+            let selectors = expr.clone();
+            let schema = schema.clone();
+            let input = lower_ir(*input, ir_arena, expr_arena, phys_sm)?;
+            Ok(phys_sm.insert(PhysNode::Select {
+                input,
+                selectors,
+                schema,
+                extend_original: false,
+            }))
+        },
+
+        // TODO: split partially streamable selections to avoid fallback as much as possible.
+        IR::HStack {
+            input,
+            exprs,
+            schema,
+            ..
+        } if exprs.iter().all(|e| is_streamable(e.node(), expr_arena)) => {
+            let selectors = exprs.clone();
+            let schema = schema.clone();
+            let input = lower_ir(*input, ir_arena, expr_arena, phys_sm)?;
+            Ok(phys_sm.insert(PhysNode::Select {
+                input,
+                selectors,
+                schema,
+                extend_original: true,
+            }))
+        },
+
```

Comment:
Do we miss a `SimpleProjection` here?

Reply:
Yes, I can add it as an optimization in a follow-up PR.

Reply:
Oh I didn't realize it was already part of the IR, added it now.